compiler/src/linking/link.re

```diff
@@ -46,6 +45,17 @@ let is_grain_module = mod_name => {
   Str.string_match(Str.regexp_string("GRAIN$MODULE$"), mod_name, 0);
 };
 
+let wasi_polyfill_module = () => {
+  "GRAIN$MODULE$./"
+  ++ Filename.remove_extension(Option.get(Config.wasi_polyfill^));
+};
+
+let is_wasi_module = mod_name => {
+  mod_name == "wasi_snapshot_preview1";
+};
+
+let is_wasi_polyfill_module = mod_name => mod_name == wasi_polyfill_module();
```

Comment:
😬  I see why we have to do this, but I think we should open an issue to remove this one day...I'm concerned about fragility in the long-term, and ideally we could just have the polyfill be a `.gr.wasm` file instead of the `.gr` source (and then Silo will bring it all together smoothly)
```suggestion

// [TODO] When silo is implemented, remove this logic
let is_wasi_polyfill_module = mod_name => mod_name == wasi_polyfill_module();
```

Reply:
Technically we don't have to do this if we don't want to support making wasi calls from the polyfills—it's _really_ only here because there isn't really a better way to write a meaningful test, haha. I feel like it'd be rare to do wasi calls from the polyfill, since if you had wasi, you wouldn't use the polyfill.

I have mixed feelings about it being a `.gr.wasm` file, since we can't have the compiler force you into runtime mode, for example. It also sort of makes it unclear what'd be allowed as a polyfill, since a fully linked module wouldn't be allowed, for example. That does bring back the idea of different file extensions for linked/unlinked modules though as a sort of solution.

Reply:
> I feel like it'd be rare to do wasi calls from the polyfill, since if you had wasi, you wouldn't use the polyfill.

It might be something we do in fact want to support, though, for reasons similar to `LD_PRELOAD` in Linux, so I guess it makes sense...I'm not going to hold up this PR on it, but I am concerned about having conditional logic based on is-this-the-polyfill in the linker.

> That does bring back the idea of different file extensions for linked/unlinked modules though as a sort of solution.

I think this would be the most pragmatic long-term solution to this issue.